hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -168,21 +168,21 @@ public class HoodieWriteConfig extends HoodieConfig {
 
   public static final ConfigProperty<String> WRITE_PAYLOAD_CLASS_NAME = ConfigProperty
       .key("hoodie.datasource.write.payload.class")
-      .defaultValue(DefaultHoodieRecordPayload.class.getName())
+      .noDefaultValue()
```

Comment:
We may still need the default value for backwards compatible write on 0.x tables cc @vinothchandar @bvaradar @balaji-varadarajan

Reply:
I don't think we do. But we might need to add something so that it doesn't remove the payload class from hoodie.properties

Reply:
The point is that, for the backwards compatible writer writing table version 6 (0.14 release and above) in Hudi 1.0 release, it should not use the merge mode and only leverage payload class to create the table.

Reply:
Let's create a Jira ticket to track the backwards compatible reader based on default payload class.

Reply:
https://issues.apache.org/jira/browse/HUDI-8408